### ISSUE TITLE
chore: update `pyproject.toml`; maybe migrate Poetry -> Hatch

### ISSUE THREAD

**Issue by OwenPriceSkelly** [Garden-AI/garden#128]
_Update: I was seeing the wrong version because I was testing from an environment which had installed `garden-ai` from git, not pypi. For the use-case of pinning ourselves as a pipeline dependency, the sdk would be able to infer its own correct version assuming users installed from pypi, which is a pretty safe assumption. Regardless, some package metadata is outdated (authors, license etc) and should be updated even if we don't throw poetry out with the bathwater. Keeping the rest of the story here for future reference; this soured me a bit on poetry and maybe it'll help me hold a grudge_

_As a pipeline, I have `garden-ai` as a dependency and need to know what its version is at runtime. Currently, that looks like the static `"0.1.0"` in `pyproject.toml`_

tl;dr: we can only have one: `poetry`, or a `pyproject.toml` in the repo that does not misrepresent the current version of the project. 
## The Problem:
  - https://xkcd.com/927/, but for python packaging tools that use `pyproject.toml` 
  - Per #63, the version which lives statically in `pyproject.toml` is indeed still wrong, yet the version which is [sent to PyPI](https://github.com/Garden-AI/garden/blob/9b26e3d91904605cc56d6a13a64174db29ee0258/.github/workflows/pypi.yaml#L22) is correct. This is because we use `poetry version` to modify the `pyproject.toml` just before sending it off to PyPI, which makes me uncomfy, as that modified `pyproject.toml` (the only one with a correct version) never appears in our repo.
  - Poetry is [not planning to support](https://github.com/python-poetry/poetry/issues/4971) setting a version dynamically, unless you also specify a static version for it to ignore. 

## The Solution(s):
0. set the version statically
    - I don't like this option. I think (philosophically) that a version is really only meaningful insofar as it corresponds to an actual release; if releases are unavoidably going to be _a_ source of truth, just let them be _the_ source of truth
1. Set the version dynamically, and lie to `pyproject.toml` about it 
    - this is what we currently do with poetry 
    -  We only ever set the version dynamically with GH actions, so I'd much prefer an honest `dynamic = ["version"]` in `pyproject.toml` to a dummy `version = "0.1.0"` that we are required to specify, just in order to ignore. 
2. Set the version dynamically, and tell `pyproject.toml` that's what we're doing
    - this would require us to move to a different build tool, like hatch 
    - The relevant PEP [621](https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#declaring-project-metadata) explicitly allows this use case
    - Of the seven (7) only-one-way-to-do-its that [PyPA recommend](https://packaging.python.org/en/latest/guides/single-sourcing-package-version/)s for single-sourcing versions, the top recommendation is to do it like this (i.e. declare the `"version"` metadata field as dynamic, so there are no misleading version numbers floating around) 

## Why hatch?
- Seems like good adoption https://hatch.pypa.io/latest/community/users/ 
  - especially sold on the fastAPI cameo, since I think the next best poetry-based solution was to use a poetry [extension](https://github.com/tiangolo/poetry-version-plugin) tiangolo wrote a couple years ago 
- seems like minimal lock-in if it doesn't fit 

# Acceptance Criteria 
- todo
